### PR TITLE
fix(driver): fix moveToStart/End, add tests

### LINKED ISSUE #5502: add tests for {moveToStart} / {moveToEnd} special sequences
add better tests for {moveToStart} / {moveToEnd} special char sequences

they were added in #4870 but not called out in changelog / documented / or tested, making it one of those "undocumented features")

* after this is fixed, we can open PR for adding to docs

### LINKED ISSUE #8465: cy.type moveToStart/End in div[contenteditable] only moves to start/end of line, not entire content
### Current behavior:
In cy.type when using `{moveToStart/End}` inside of a `div[contenteditable]`, the cursor is only moved to the start/end of the line, not the entire text content.

for example when using `{moveToStart}` 
expected:
```
foo      ->    Ifoo
bar			   bar
bazI		   baz
```
actual:
```
foo      ->    foo
bar			   bar
bazI		   Ibaz
```

### Versions
3.5 +

### PR DESCRIPTION
**Pull request by kuceb**
cont. of #7970 to move it to an internal branch
 
- fix #8465
- Closes #5502 

### User facing changelog
Bug fix: fix issue with `cy.type` `{moveToStart}`/`{moveToEnd}` special characters moving cursor on the current line instead of the entire text editable when typing in a `contentEditable` element.

### Additional details
- thanks to @sainthkh for adding tests for moveToStart/End and surfacing this bug.

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by Bkucera** [comment]
@sainthkh let me know what you think, used a simple selection.collapse for chrome, had to add more logic for firefox to fix the selection to prevent newline issue

File: packages/driver/src/dom/selection.ts [hunk 1]
```diff
@@ -563,20 +556,44 @@ const _moveSelectionTo = function (toStart: boolean, el: HTMLElement, options =
...
+
+      range.selectNodeContents(elToSelect)
+    } else {
+      $elements.callNativeMethod(doc, 'execCommand', 'selectAll', false, null)
+    }
+
+    toStart ? selection.collapseToStart() : selection.collapseToEnd()
   }
 
   return false
```
Comment (sainthkh) [inline packages/driver/src/dom/selection.ts:599]: We don't need this line. Every `return` above returns `undefined` but it's `false` here. And `moveSelectionToEnd` and `moveSelectionToStart` doesn't need a return value.

File: packages/driver/src/dom/selection.ts [hunk 2]
```diff
@@ -563,20 +556,44 @@ const _moveSelectionTo = function (toStart: boolean, el: HTMLElement, options =
...
+      // FireFox doesn't treat a selectall+arrow the same as clicking the start/end of a contenteditable
+      // so we need to select the specific nodes inside the contenteditable.
+      const root = getHostContenteditable(el)
 
-    selection.modify('move', direction, 'line')
+      let elToSelect = root.childNodes[toStart ? 0 : root.childNodes.length - 1]
 
-    return
+      // in firefox, an empty contenteditable my be a single <br> element, or <div><br/></div>
+      // but its innerText will be '\n' (maybe find a more efficient measure)
```
Comment (sainthkh) [inline packages/driver/src/dom/selection.ts:573]:
```suggestion
      // in firefox, when an empty contenteditable is a single <br> element, or <div><br/></div>
      // its innerText will be '\n' (maybe find a more efficient measure)
```

`my be` might be a typo of `might be` or `may be`

**Review by sainthkh — CHANGES_REQUESTED** [review]
LGTM. There are only 2 minor changes.

**Pull request merged**